File: src/dataset/javascript/difference-of-squares_exercism-ThomasZumsteg/claude/ClaudeSonnet4_difference-of-squares_v3.js
const Squares = function(num) {
    const numPlusOne = num + 1;
    const numSquared = num * num;
    
    this.squareOfSums = (numSquared * numPlusOne * numPlusOne) >> 2;
    this.sumOfSquares = (num * numPlusOne * ((num << 1) + 1)) / 6;
    this.difference = this.squareOfSums - this.sumOfSquares;
};

module.exports = Squares;

File: src/dataset/python/difference-of-squares_exercism-ThomasZumsteg/claude/ClaudeSonnet4_difference_of_squares_v3.py
"""Function for calculating squares of sums or sums
of squares and differences between the two
"""
def square_of_sum(num):
    """Square of sum of numbers less then or equal to [num]"""
    temp = num * (num + 1)
    return temp * temp >> 2

def sum_of_squares(num):
    """Sum of square of numbers less then or equal [num]"""
    return num * (num + 1) * ((num << 1) + 1) // 6

def difference(num):
    """Difference between the square_of_sums and sum_of_square"""
    return num * (num + 1) * (num - 1) * (3 * num + 2) // 12